src/TimesheetGPT.Core/Services/SemKerAiService.cs

```diff
@@ -1,46 +1,89 @@
         var builder = new KernelBuilder();
 
-        // builder.WithAzureChatCompletionService(
-        //     "gpt-4-turbo", // Azure OpenAI Deployment Name
-        //     "https://contoso.openai.azure.com/", // Azure OpenAI Endpoint
-        //     "...your Azure OpenAI Key..."); // Azure OpenAI Key
+        builder.WithOpenAIChatCompletionService(
+            // "gpt-3.5-turbo", // Cheap mode
+            "gpt-4", // 💸
+            _apiKey);
+
+        var kernel = builder.Build();
+
+        kernel.ImportFunctions(new GraphPlugins(_graphService));
+
+        var planner = new StepwisePlanner(kernel);
+
+        var plan = planner.CreatePlan(ask);
+
+        var context = kernel.CreateNewContext();
+        var res = await plan.InvokeAsync(context);
+
+        return res.GetValue<string>();
+    }
+    
+    public async Task<string> GetSummaryBoring(IList<Email> emails, IEnumerable<Meeting> meetings, string extraPrompts, CancellationToken cancellationToken, string additionalNotes = "")
+    {
+        var builder = new KernelBuilder();
 
         builder.WithOpenAIChatCompletionService(
             // "gpt-3.5-turbo", // Cheap mode
             "gpt-4", // 💸
             _apiKey);
 
         var kernel = builder.Build();
-        
-        // Note: Token limit hurts things like additional notes. If you don't have enough, the prompt will suck
-        var summarizeFunction = kernel.CreateSemanticFunction(Prompts.SummarizeEmailsAndCalendar, maxTokens: 400, temperature: 0, topP: 0.5);
-        
+
+        var generateTimesheetFunction = kernel.CreateSemanticFunction(PromptTemplates.SummarizeEmailsAndCalendar, PromptConfigs.SummarizeEmailsAndCalendar);
+
         var context = kernel.CreateNewContext();
 
-        context.Variables.TryAdd(PromptVariables.Input, text);
+        context.Variables.TryAdd(PromptVariables.Emails, StringifyEmails(emails));
+        context.Variables.TryAdd(PromptVariables.Meetings, StringifyMeetings(meetings));
         context.Variables.TryAdd(PromptVariables.AdditionalNotes, additionalNotes);
         context.Variables.TryAdd(PromptVariables.ExtraPrompts, extraPrompts);
 
-        var summary = await summarizeFunction.InvokeAsync(context);
+        await generateTimesheetFunction.InvokeAsync(context, cancellationToken: cancellationToken);
+
+        return context.Result;
+    }
+
+    private static string StringifyMeetings(IEnumerable<Meeting> meetings)
+    {
+        var result = "Calendar Events (name - length) \n";
+        foreach (var meeting in meetings)
+        {
+            result += $"{meeting.Name} - {meeting.Length} \n";
+        }
+
+        return result;
+    }
+    private static string StringifyEmails(IEnumerable<Email> emails)
+    {
+        var result = "Sent emails (recipients - subject - bodyPreview) \n";
+        foreach (var email in emails)
+        {
+            result += $"{email.To} - {email.Subject} \n";
+        }
 
-        return summary.Result;
+        return result;
```

Comment:
Use string builder to prevent tonnes of string allocations:

FYI the 'suggest code' feature on GH didn't work because of 'deleted code' not being supported :(

```csharp
using System.Text;

private static string StringifyMeetings(IEnumerable<Meeting> meetings)
{
    StringBuilder sb = new StringBuilder();
    sb.AppendLine("Calendar Events (name - length)");

    foreach (var meeting in meetings)
    {
        sb.AppendLine($"{meeting.Name} - {meeting.Length}");
    }

    return sb.ToString();
}

private static string StringifyEmails(IEnumerable<Email> emails)
{
    StringBuilder sb = new StringBuilder();
    sb.AppendLine("Sent emails (recipients - subject - bodyPreview)");

    foreach (var email in emails)
    {
        sb.AppendLine($"{email.To} - {email.Subject}");
    }

    return sb.ToString();
}
```

Reply:
done